src/main/java/org/datadog/jenkins/plugins/datadog/clients/HttpClient.java

```diff
@@ -0,0 +1,291 @@
+                result.setDaemon(true);
+                return result;
+            }
+        };
+
+        QueuedThreadPool threadPool = new QueuedThreadPool(
+                64,
+                1,
+                60_000,
+                -1,
+                queue,
+                null,
+                threadFactory
+        );
+        threadPool.setName("dd-http-client-thread-pool");
+
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
+        org.eclipse.jetty.client.HttpClient httpClient = new org.eclipse.jetty.client.HttpClient(sslContextFactory);
+        httpClient.setExecutor(threadPool);
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not start HTTP client", e);
+        }
+        return httpClient;
+    }
+
+    private static void ensureProxyConfiguration() {
+        Jenkins jenkins = Jenkins.getInstanceOrNull();
+        if (jenkins == null) {
+            return;
+        }
+        hudson.ProxyConfiguration jenkinsProxyConfiguration = jenkins.getProxy();
+        if (jenkinsProxyConfiguration == null) {
+            return;
+        }
+
+        if (EFFECTIVE_PROXY_CONFIGURATION != jenkinsProxyConfiguration) {
+            synchronized (CLIENT) {
+                if (EFFECTIVE_PROXY_CONFIGURATION != jenkinsProxyConfiguration) {
+                    org.eclipse.jetty.client.ProxyConfiguration proxyConfig = CLIENT.getProxyConfiguration();
+                    List<ProxyConfiguration.Proxy> proxies = proxyConfig.getProxies();
+                    proxies.clear();
+
+                    String proxyHost = jenkinsProxyConfiguration.getName();
+                    int proxyPort = jenkinsProxyConfiguration.getPort();
+                    List<Pattern> noProxyHostPatterns = jenkinsProxyConfiguration.getNoProxyHostPatterns();
+                    proxies.add(new HttpProxy(proxyHost, proxyPort) {
+                        @Override
+                        public boolean matches(Origin origin) {
+                            Origin.Address address = origin.getAddress();
+                            String host = address.getHost();
+                            for (Pattern noProxyHostPattern : noProxyHostPatterns) {
+                                if (noProxyHostPattern.matcher(host).matches()) {
+                                    return false;
+                                }
+                            }
+                            return true;
+                        }
+                    });
+
+                    EFFECTIVE_PROXY_CONFIGURATION = jenkinsProxyConfiguration;
+                }
+            }
+        }
+    }
+
+    private final long timeoutMillis;
+    private final HttpRetryPolicy.Factory retryPolicyFactory;
+
+    public HttpClient(long timeoutMillis) {
+        this.timeoutMillis = timeoutMillis;
+        this.retryPolicyFactory = new HttpRetryPolicy.Factory(5, 100, 2.0);
```

Comment:
Can we have these parameters as constants somewhere? I'm also wondering if it would be helpful for them to be configurable to users, maybe just through environment variables

Reply:
Agree, moved all the magic numbers to named constants and provided means of overriding defaults via env vars

Reply:
Nice, thanks!